src/graphql/resolvers/queries/communities.js

```diff
@@ -60,5 +61,50 @@ export const fieldResolvers = {
         totalEvents: allEvents.length,
       };
     },
+    sendDigest: async (
+      { name },
+      { hours, start },
+      { dataSources: { firestore } },
+    ) => {
+      dlog('sendDialog called for %s, hours: %s', name, hours);
+      if (!hours) throw new Error('hours parameter required');
+      if (hours < 1) throw new Error('hours minimum value is 1');
+      if (hours > 168) throw new Error('hours maximum value is 168');
+      const activeEvents = await communityStore(firestore).findIsActiveEvents(
+        name,
+      );
+      let digestStart = 'CURRENT_HOUR';
+      if (start) digestStart = start;
+
+      let atDate;
+      // Date as of now min, sec, ms set to zero
+      if (digestStart === 'CURRENT_HOUR') {
+        atDate = new Date(new Date(Date.now()).setMinutes(0, 0, 0));
+      } else if (digestStart === 'NEXT_HOUR') {
+        // now + 1 hour (3600000 ms)
+        atDate = new Date(
+          new Date(Date.now()).setMinutes(0, 0, 0) + 60 * 60 * 1000,
+        );
+      } else {
+        throw new Error(`Unknown value sent for 'start': ${digestStart}`);
+      }
+      const hoursAfter = hours || 0;
+      const sessionFuncs = activeEvents.map(ev =>
+        sessionStore(firestore).findAllApprovedByEventIdAtDateHours(
+          ev.id,
+          atDate,
+          hoursAfter,
+        ),
+      );
+      const sessionRefs = await Promise.all(sessionFuncs);
+      const sessions = [];
+      sessionRefs.forEach(s => sessions.push(...s));
```

Comment:
why not just use .map? and even having said that.. not sure I get why take one and just push into another?

Reply:
Map will return a single array in this case? I am taking 1 + N array of objects and making them into one array of objects.

Reply:
I gotcha..